### PR TITLE
[MBO-1524] Add service modules helper + Implement method to find updates for a module

### PR DESCRIPTION
**Pull request by sowbiba**
## Usage

```
$this->get('mbo.modules.helper')->findForUpdates('productcomments')
```

will return

```
[
  "current_version" => "5.0.2"
  "available_version" => "6.0.2"
  "upgrade_available" => true,
  'urls' => [
      'install' => null,
      'upgrade' => '/manage/action/upgrade/productcomments',
   ]
]
```
If the module is installed and do have an upgrade

============================================================

```
[
  "current_version" => "5.0.2"
  "available_version" => "5.0.2"
  "upgrade_available" => false,
  'urls' => [
      'install' => null,
      'upgrade' => null,
   ]
]
```
If the module is installed but doesn't have an upgrade

============================================================

```
[
  "current_version" => null
  "available_version" => "5.0.2"
  "upgrade_available" => false,
  'urls' => [
      'install' => '/manage/action/install/productcomments',
      'upgrade' => null,
   ]
]
```
If the module is not installed but exists on Addons

============================================================

```
[
  "current_version" => null
  "available_version" => null
  "upgrade_available" => false,
  'urls' => [
      'install' => null,
      'upgrade' => null,
   ]
]
```
If the module is not installed and doesn't exists on Addons

### CODE REVIEW

**Comment by intraordinaire** [comment]
Looks great !

File: src/Service/ModulesHelper.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        /** @var string|false $moduleCurrentVersion */
+        $moduleCurrentVersion = $db->getValue($request);
+
+        if ($moduleCurrentVersion) {
+            $currentVersion = $moduleCurrentVersion;
+        }
+
+        // We need to clear cache to get fresh data from addons
+        $this->moduleRepository->clearCache();
```
Comment (intraordinaire) [inline src/Service/ModulesHelper.php:55]: Do we clear cache on each call ? Or reduce the cache time to, let's say, to 1h ?
If someone call this function on header or whatever, the cache will be cleared each time.

**Review by intraordinaire — APPROVED** [review]

**Comment by intraordinaire** [comment]
@sowbiba 
Maybe add a comment about this in the Readme ?
And a usage with a try/catch to load the service ?

**Comment by sowbiba** [comment]
Done ✔️ 

https://github.com/PrestaShopCorp/ps_mbo/blob/feat/MBO-1524/modules-helper-find-updates/README.md#tools-and-helpers

**Review by intraordinaire — APPROVED** [review]

**Pull request merged**